core/irCheck.ml

```diff
@@ -965,17 +1020,17 @@ struct
 
 
           (* We now construct the inner effects from the outer effects and branch_presence_spec_types *)
-          let (outer_effects_map, outer_effects_var, outer_effects_dualized) = outer_effects in
+          let (outer_effects_map, outer_effects_var, outer_effects_dualized) = outer_effects_parts in
           (* For each case branch, the corresponding entry goes directly into the field spec map of the inner effect row *)
-          let inner_effects_map_from_branches = StringMap.map (fun x -> `Present x) branch_presence_spec_types in
+          let inner_effects_map_from_branches = StringMap.map (fun x -> Present x) branch_presence_spec_types in
           (* We now add all entries from the outer effects that were not touched by the handler to the inner effects *)
           let inner_effects_map = StringMap.fold (fun effect outer_presence_spec map ->
               if StringMap.mem effect inner_effects_map_from_branches then
                 map
               else
                 StringMap.add effect outer_presence_spec map
             )  inner_effects_map_from_branches outer_effects_map in
-          let inner_effects = (inner_effects_map, outer_effects_var, outer_effects_dualized) in
+          let inner_effects = Row (inner_effects_map, outer_effects_var, outer_effects_dualized) in
```

Comment:
Just curious why `Row` here and not `Effect`?

Reply:
Well we need a row here anyway... really we just need a triple of type `row'`. We couldn't do `Effect ...` on its own but we could do `Effect (Row ...)` which would be even more annoying than what we have now with the `Row` constructor as it would require a further level of boxing/unboxing. That said, it might be worth revisiting this when we rationalise the design further, and perhaps it should morally be `Effect (Row ...)`.